spec/lib/sendgrid_helper_spec.rb

```diff
@@ -2,62 +2,91 @@
-        it "marks the email as valid if there's an error (#{exception_class})" do
-          expect(Curl::Easy).to receive(:perform).and_raise(exception_class)
+  context 'with sendgrid configured' do
+    context "spam reports" do
+      context "error handling" do
+        [Curl::Err::CurlError, JSON::ParserError].each do |exception_class|
+          it "marks the email as valid if there's an error (#{exception_class})" do
+            allow(Curl::Easy).to receive(:perform).and_raise(exception_class)
+            expect(SendgridHelper.spam_reported?('test@irrelevant.com')).to be_falsey
+          end
         end
-      end
 
-      it "marks the email as valid when authentication fails" do
-        allow(JSON).to receive(:parse).and_return({error: 'lol'})
+        it "marks the email as valid when authentication fails" do
+          allow(JSON).to receive(:parse).and_return({error: 'lol'})
+          expect(SendgridHelper.spam_reported?('test@irrelevant.com')).to be_falsey
+        end
       end
 
-      after :each do
-        expect(SendgridHelper.spam_reported?('test@irrelevant.com')).to be_falsey
+      context "when no error" do
+        it "marks an e-mail as valid" do
+          expect(JSON).to receive(:parse).and_return(['dummy'])
+          SendgridHelper.spam_reported?('test@irrelevant.com')
+        end
+
+        it "marks an e-mail as invalid" do
+          expect(JSON).to receive(:parse).and_return([])
+          SendgridHelper.spam_reported?('test@irrelevant.com')
+        end
       end
     end
 
-    context "when no error" do
-      it "marks an e-mail as valid" do
-        expect(JSON).to receive(:parse).and_return(['dummy'])
-        SendgridHelper.spam_reported?('test@irrelevant.com')
+    context "bounced" do
+      context "error handling" do
+        [Curl::Err::CurlError, JSON::ParserError].each do |exception_class|
+          it "marks the email as valid if there's an error (#{exception_class})" do
+            allow(Curl::Easy).to receive(:perform).and_raise(exception_class)
+            expect(SendgridHelper.bounced?('test@irrelevant.com')).to be_falsey
+          end
+        end
+
+        it "marks the email as valid when authentication fails" do
+          allow(JSON).to receive(:parse).and_return({error: 'lol'})
+          expect(SendgridHelper.bounced?('test@irrelevant.com')).to be_falsey
+        end
       end
 
-      it "marks an e-mail as invalid" do
-        expect(JSON).to receive(:parse).and_return([])
-        SendgridHelper.spam_reported?('test@irrelevant.com')
+      context "when no error" do
+        it "marks an e-mail as valid" do
+          expect(JSON).to receive(:parse).and_return(['dummy'])
+          SendgridHelper.bounced?('test@irrelevant.com')
+        end
+
+        it "marks an e-mail as invalid" do
+          expect(JSON).to receive(:parse).and_return([])
+          SendgridHelper.bounced?('test@irrelevant.com')
```

Comment:
I don't think these tests are actually evaluating truthiness of the return of `bounced?`, are they @threedaymonk?